src/GoTrueClient.ts

```diff
@@ -634,7 +635,7 @@ export default class GoTrueClient {
    */
   private async _recoverAndRefresh() {
     try {
-      const json = isBrowser() && (await this.localStorage.getItem(STORAGE_KEY))
+      const json = isBrowser() && (await this.localStorage?.getItem(STORAGE_KEY))
```

Comment:
extract localstorage.(get, set, removeitem) into a utility cos we seem to be doing isbrowser() everytime before accessing it and will lead to errors if we dont do that guard. (can be in a separate PR)